createtransfers/createtransfers.py

```diff
@@ -162,15 +168,15 @@ def stat_file(file_path, files_n):
                   * files_n) / 1024.0) / 1024.0 / 1024.0)
 
 
-def create_variously_encoded_files():
+def create_variously_encoded_files(zip_path=None):
     """Create files with strange (non-UTF8) encodings under
     TestTransfers/files_with_various_encodings/
     """
-    target_path = os.path.join(
-        HERE, 'TestTransfers', 'files_with_various_encodings')
-
+    sub_dir = 'files_with_various_encodings'
+    target_path = os.path.join(HERE, 'TestTransfers', sub_dir)
+    if zip_path:
+        target_path = os.path.join(zip_path)
```

Comment:
Why are you calling `os.path.join` on something that is already a path? This happens several times in the newly introduced code. I think all occasions should be removed.

Reply:
Not good! Weird thing to do first and then copy and paste mostly. :)